rclcpp/include/rclcpp/service.hpp

```diff
@@ -172,6 +172,7 @@ class Service : public ServiceBase
         (std::cerr << ss.str()).flush();
         rcl_reset_error();
       }
+      delete service_handle_;
```

Comment:
Sorry, I'm not sure I understand.  This `delete service_handle_` is inside of the `if (owns_rcl_handle_)` block, so it will only be freed if we own the handle.  Did you mean something else?